freshmaker/lightblue.py

```diff
@@ -1825,3 +1830,50 @@ def get_fixed_published_image(self, name, version, image_group, rpm_nvrs, conten
         image = images[0]
         image.resolve(self)
         return image
+
+    def is_latest_eus_image(self, image):
+        """
+        Check if the image is EUS base image and his repository marked with
+        auto_rebuild tag. EUS repository contains '-els' suffix.
+
+        :param ContainerImage image: image to check
+        :param dict repo_to_auto_rebuild_tags: mapping of processed repositories
+            names to set of their 'auto_rebuild_tags'
+        :return: True if image is EUS base image with 'auto_rebuild_tag', False otherwise
+        """
+        def get_repo(repository):
+            query = {
+                "objectType": "containerRepository",
+                "query": {
+                    "$and": [
+                        {
+                            "field": "repository",
+                            "op": "=",
+                            "rvalue": repository
+                        },
+                    ]
+                },
+                "projection": [
+                    {"field": "auto_rebuild_tags", "include": True, "recursive": True}
+                ]
+            }
+            repos = self.find_container_repositories(query, auto_rebuild=True)
+            return repos[0]
+
+        for repo in image["repositories"]:
+            repo_name = repo["repository"]
+            # Check tags only if it's EUS image
+            if re.match(r".*-els($|\W)", repo_name):
+                auto_rebuild_tags = self.repo_to_auto_rebuild_tags.get(repo_name, None)
```

Comment:
Optional: `None` is not needed. It's the default.
```suggestion
                auto_rebuild_tags = self.repo_to_auto_rebuild_tags.get(repo_name)
```